### PR TITLE
Add list method for answering multi-file upload questions

### PR DESCRIPTION
**Pull request by devin-civi**
### Description

Following up from previous PR. Adding option to answer file uploads with a list, instead of index by index. 

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)

### CODE REVIEW

**Comment by devin-civi** [comment]
* **#8383** <a href="https://app.graphite.dev/github/pr/civiform/civiform/8383?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @devin-civi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: server/test/services/applicant/ReadOnlyApplicantProgramServiceImplTest.java [hunk 1]
```diff
@@ -124,14 +124,7 @@ public void getStoredFileKeys_worksForMultipleFileUploads() {
...
         ApplicantData.APPLICANT_PATH.join(fileQuestionDefinition.getQuestionPathSegment()),
-        0,
-        "file-key");
-
-    QuestionAnswerer.answerFileQuestionWithMultipleUpload(
-        applicantData,
-        ApplicantData.APPLICANT_PATH.join(fileQuestionDefinition.getQuestionPathSegment()),
-        1,
-        "file-key-2");
+        ImmutableList.of("file-key", "file-key-2"));
```
Comment (melanie-exygy) [inline server/test/services/applicant/ReadOnlyApplicantProgramServiceImplTest.java:127]: **praise:** I'm loving that you're reducing all these lines of code down to 1.  So satisfying!

**Review by melanie-exygy — APPROVED** [review]

File: server/app/services/question/QuestionAnswerer.java [hunk 2]
```diff
@@ -77,6 +78,15 @@ public static void answerFileQuestionWithMultipleUpload(
         contextualizedPath.join(Scalar.FILE_KEY_LIST + Path.ARRAY_SUFFIX).atIndex(index), fileKey);
   }
 
+  public static void answerFileQuestionWithMultipleUpload(
+      ApplicantData applicantData, Path contextualizedPath, List<String> fileKeys) {
```
Comment (avaleske) [inline server/app/services/question/QuestionAnswerer.java:82]: Can we update this to take an `ImmutableList`?

File: server/test/services/export/AbstractExporterTest.java [hunk 3]
```diff
@@ -968,12 +969,12 @@ FakeApplicationFiller answerFileUploadQuestion(
     }
 
     FakeApplicationFiller answerFileQuestionWithMultipleUpload(
-        QuestionModel question, String fileKey, int index) {
-      return answerFileQuestionWithMultipleUpload(question, null, fileKey, index);
+        QuestionModel question, List<String> fileKeys) {
```
Comment (avaleske) [inline server/test/services/export/AbstractExporterTest.java:972]: ditto here re `ImmutableList`
https://github.com/civiform/civiform/wiki/Development-standards#prefer-using-immutable-collection-types

**Review by avaleske — APPROVED** [review]
Thank you!

**Pull request merged**